File: CS/FilterReportParameter.Module/BusinessObjects/TestParameterClass.cs
using System;
using System.Linq;
using DevExpress.Xpo;
using System.ComponentModel;
using DevExpress.Persistent.Base;
using System.Collections.Generic;
using DevExpress.Persistent.BaseImpl;

namespace FilterReportParameter.Module.BusinessObjects {
    [DefaultClassOptions]
    [DefaultProperty("Name")]
    public class DemoParameter : BaseObject {
        public DemoParameter(Session session)
            : base(session) {
        }
        public string Name {
            get { return GetPropertyValue<string>("Name"); }
            set { SetPropertyValue("Name", value); }
        }
        public string Category {
            get { return GetPropertyValue<string>("Category"); }
            set { SetPropertyValue("Category", value); }
        }
    }
}

File: VB/FilterReportParameter.Module/BusinessObjects/TestParameterClass.vb
Imports System
Imports System.Linq
Imports DevExpress.Xpo
Imports System.ComponentModel
Imports DevExpress.Persistent.Base
Imports System.Collections.Generic
Imports DevExpress.Persistent.BaseImpl

Namespace FilterReportParameter.Module.BusinessObjects
    <DefaultClassOptions, DefaultProperty("Name")> _
    Public Class DemoParameter
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Property Name() As String
            Get
                Return GetPropertyValue(Of String)("Name")
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Name", value)
            End Set
        End Property
        Public Property Category() As String
            Get
                Return GetPropertyValue(Of String)("Category")
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Category", value)
            End Set
        End Property
    End Class
End Namespace
